processor/probabilisticsamplerprocessor/logsprocessor.go

```diff
@@ -0,0 +1,111 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package probabilisticsamplerprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/probabilisticsamplerprocessor"
+
+import (
+	"context"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+)
+
+type logSamplerProcessor struct {
+	scaledSamplingRate uint32
+	hashSeed           uint32
+	traceIDEnabled     bool
+	samplingSource     string
+	samplingPriority   string
+}
+
+// newLogsProcessor returns a processor.LogsProcessor that will perform head sampling according to the given
+// configuration.
+func newLogsProcessor(ctx context.Context, set component.ProcessorCreateSettings, nextConsumer consumer.Logs, cfg *Config) (component.LogsProcessor, error) {
+
+	lsp := &logSamplerProcessor{
+		scaledSamplingRate: uint32(cfg.SamplingPercentage * percentageScaleFactor),
+		hashSeed:           cfg.HashSeed,
+		traceIDEnabled:     cfg.AttributeSource == traceIDAttributeSource,
+		samplingPriority:   cfg.SamplingPriority,
+		samplingSource:     cfg.FromAttribute,
+	}
+
+	return processorhelper.NewLogsProcessor(
+		ctx,
+		set,
+		cfg,
+		nextConsumer,
+		lsp.processLogs,
+		processorhelper.WithCapabilities(consumer.Capabilities{MutatesData: true}))
+}
+
+func (lsp *logSamplerProcessor) processLogs(ctx context.Context, ld plog.Logs) (plog.Logs, error) {
+	ld.ResourceLogs().RemoveIf(func(rl plog.ResourceLogs) bool {
+		rl.ScopeLogs().RemoveIf(func(ill plog.ScopeLogs) bool {
+			ill.LogRecords().RemoveIf(func(l plog.LogRecord) bool {
+
+				tagPolicyValue := "always_sampling"
+				// pick the sampling source.
+				var lidBytes []byte
+				if lsp.traceIDEnabled && !l.TraceID().IsEmpty() {
+					value := l.TraceID()
+					tagPolicyValue = "trace_id_hash"
+					lidBytes = value[:]
+				}
+				if lidBytes == nil && lsp.samplingSource != "" {
+					if value, ok := l.Attributes().Get(lsp.samplingSource); ok {
+						tagPolicyValue = lsp.samplingSource
+						lidBytes = value.Bytes().AsRaw()
+					}
+				}
+				priority := lsp.scaledSamplingRate
+				if lsp.samplingPriority != "" {
+					if localPriority, ok := l.Attributes().Get(lsp.samplingPriority); ok {
```

Comment:
Spans can be sampled based on the `sampling.priority` attribute. We intend to do similar for logs data.
I think it can be a good starting point for adding logs sampling.